packages/mui-material-next/src/Button/Button.tsx

```diff
@@ -490,12 +538,16 @@ const Button = React.forwardRef(function Button<
       as={ComponentProp}
       className={clsx(classes.root, className)}
       ownerState={ownerState}
-      {...getRootProps()}
+      {...getRootProps(getRippleHandlers(props) as unknown as EventHandlers)}
```

Comment:
I suppose get*Props in Base hooks should be more permissive. Note taken.

Reply:
Ideally yes :)